Automattic/wp-calypso
# DIFM: Fix new site link on site picker page

#### Proposed Changes

Bug Report: p1661952943943549-slack-C02KVCAL7GX

* Due to missing dependencies, the new site link added in #66934 did not work. This PR fixes the behaviour of the new site link by dispatching the `submitSignupStep` with the required dependencies.

#### Testing Instructions



* Go to `/start/do-it-for-me` and choose Existing Site.
* Click on the "new site" link.
<img width="1005" alt="image" src="https://user-images.githubusercontent.com/5436027/188085989-bd34930f-38d9-4850-be67-ce52664e6bd5.png">

* Confirm that you can proceed to checkout and complete the purchase.

**Testing for regressions**
* Go to `/start/do-it-for-me` and choose Existing Site. Select an existing site and confirm that you can complete the flow.
* Go to `/start/do-it-for-me` and choose New Site. Confirm that you can complete the flow.


#### Pre-merge Checklist






Fixes Automattic/martech#1054
